app/src/main/java/io/intrepid/contest/utils/SlidingTabAdapter.java

```diff
@@ -0,0 +1,107 @@
+package io.intrepid.contest.utils;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.intrepid.contest.base.BaseFragment;
+import timber.log.Timber;
+
+public class SlidingTabAdapter extends FragmentPagerAdapter implements ViewPager.OnPageChangeListener{
+    private final AppCompatActivity activity;
+    private int currentIndex = 0;
+    private List<BaseFragment> fragments = new ArrayList<>();
+
+    public SlidingTabAdapter(AppCompatActivity activity) {
+        super(activity.getSupportFragmentManager());
+        this.activity = activity;
+    }
+
+    @Override
+    public BaseFragment getItem(int position) {
+        return fragments.get(position);
+    }
+
+    @Override
+    public Object instantiateItem(ViewGroup container, int position) {
+        BaseFragment fragment = (BaseFragment) super.instantiateItem(container, position);
+        fragments.set(position, fragment);
+        return fragment;
+    }
+
+    @Override
+    public int getCount() {
+        return fragments.size();
+    }
+
+    public void addFragment(BaseFragment fragment) {
+        fragments.add(getCount(), fragment);
+        notifyDataSetChanged();
+    }
+
+    public int getLastItemPosition() {
+        return getCount() - 1;
+    }
+
+    public boolean isLastSlide(int position) {
+        return position == getCount() - 1;
+    }
+
+    public boolean shouldFinish(int position) {
+//        return position == getCount() && getItem(getCount() - 1).canMoveFurther();
+        return false;
+    }
+
+    public boolean canAdvance() {
+        BaseFragment fragment = getItem(currentIndex);
+//        return !fragment.canMoveFurther();
+        return true;
+    }
+
+    @Override
+    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+        currentIndex = position;
+        Timber.d("Current index " + currentIndex);
+    }
+
+    @Override
+    public void onPageSelected(int position) {
+        currentIndex = position;
+        Timber.d("Current index " + currentIndex);
+        switch (currentIndex){
+            case 0: activity.getSupportActionBar().setTitle("New Title");
+                break;
+            case 1: activity.getSupportActionBar().setTitle("Description");
+                break;
+            case 2: activity.getSupportActionBar().setTitle("");
+                break;
+            case 3: activity.getSupportActionBar().setTitle("Scoring Categories");
+                break;
+        }
+
+    }
+
+    @Override
+    public void onPageScrollStateChanged(int state) {
```

Comment:
If you remove the interface you are implementing you can remove this unused function.